# Guided Project: 
# Gun Deaths in the US (2013 to 2014) 


## Introducing the Gun Death Data

In [1]:
import csv
import datetime


# Read the dataset using the csv module
f = open('guns.csv')
csvreader = csv.reader(f)
data = list(csvreader)

print(data[:4])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4']]


###  Remove the Headers from a List of Lists

In [2]:
headers = data[0:1]
data = data[1:]
print(headers)
print(data[:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']]
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


### Counting Gun Deaths by Year

In [3]:
year_count = {}
for row in data:
    years = row[1]
    
    if years not in year_count:
        year_count[years] = 1
    else:
        year_count[years] += 1
        
year_count        


{'2012': 33563, '2013': 33636, '2014': 33599}

### Counting Gun Deaths by Year and Month

In [4]:
dates = [datetime.datetime(year=int(row[1]), month=int(row[2]), day=1) for row in data]

dates[:5]                        

                                 

[datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0)]

In [5]:
date_counts = {}
for date in dates:
    if date not in date_counts:
        date_counts[date] = 1
    else: 
        date_counts[date] += 1
        
date_counts  

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

### Counting Gun Deaths for  Gender and Race

In [6]:
genders = [row[5] for row in data]
races = [row[7] for row in data]

sex_counts = {}
race_counts = {}

for gender in genders:
    if gender not in sex_counts:
        sex_counts[gender] = 1
    else:
        sex_counts[gender] += 1

print(sex_counts)

for race in races:
    if race not in race_counts:
        race_counts[race] = 1
    else:
        race_counts[race] += 1
        
race_counts

{'M': 86349, 'F': 14449}


{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

## Mid-point Summary of Data Studied

### Gun Deaths Per Year
The gun deaths have steadily increased for the years 2012, 2013 and 2014 (36, 37).  Looking closer per month of each year there is a spike in the late spring to summer months (monthly increase pattern repeats for 2012 and 2013, it still increases for 2014, but the increase pattern is different)

### Deaths by Gender
Combining the male deaths over three years produces 86,349 compared to 14,449 deaths for women.  That's a ratio of 6 men to 1 women.

### Deaths by Race
At surface value "White" is the race with the most deaths, but 'Black' as the second largest.  This ratio is 2.8 'white' people to 1 'black' person.  #### This gives a false impression because the population of the US is not 2.8 'white' people to 1 'black' person.  The Gun-Race data should be combined and compared with the population percentage by Race in the US.

### Further Examination
Further break down of total deaths need to be examined, such as the intent type of death by gun, who shot the gun, location of shooting, and the victem's age and education.  After the general totals these columns should be used to analyse further Gender and Race (as headings with further analyses).


### Add Second Data Set: census.csv 
### Read census.csv 

In [7]:
f = open('census.csv')
csvreader = csv.reader(f)
census = list(csvreader)

census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

### Race - Population Dictionary to Match Keys 

In [8]:
mapping = {
    'Asian/Pacific Islander': 15834141,
    'Black': 40250635,
    'Native American/Native Alaskan': 3739506,
    'Hispanic': 44618105,
    'White': 197318956 
}

race_per_hundredk ={}
for k, v in race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000
        
    
race_per_hundredk    



{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.8773477735196,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

### Counting Homicide Gun Deaths for Race 

In [9]:
intents = [intent[3] for intent in data]
races = [race[7] for race in data]

homicide_race_counts = {}
for i, race in enumerate(races):
    if race not in homicide_race_counts:
        homicide_race_counts[race] = 0
    if intents[i] == 'Homicide':
        homicide_race_counts[race] += 1
        
        
homicide_race_counts
             

{'Asian/Pacific Islander': 559,
 'Black': 19510,
 'Hispanic': 5634,
 'Native American/Native Alaskan': 326,
 'White': 9147}

### Race/Homicide - Population Dictionary to Match Keys 

In [15]:
race_per_hundredk ={}
for k, v in homicide_race_counts.items():
    race_per_hundredk[k] = round((v / mapping[k]) * 100000, 2)
        
    
race_per_hundredk 

{'Asian/Pacific Islander': 3.53,
 'Black': 48.47,
 'Hispanic': 12.63,
 'Native American/Native Alaskan': 8.72,
 'White': 4.64}

## Summary of Data Studied
### Homicide Guns Deaths Per Year Based on Race 
The percentages of gun homicides deaths become more accurate with the census data added with the gun deaths data.  The highest percentage of deaths by race is African American ("Black") followed by Hispanic.

### Further Examination
The following could be analyzed more:
- Homicide rate by gender 
- Education and age of victims
- Location of shooting
- who is the shooter, police or other
- Monthly deaths, broken down by gender, age, education, location, shooter


